open-api/rest-catalog-open-api.py

```diff
@@ -267,26 +267,57 @@ class ViewVersion(BaseModel):
     default_namespace: Namespace = Field(..., alias='default-namespace')
 
 
+class ContentEnum(BaseModel):
+    __root__: Literal['data', 'equality-deletes', 'position-deletes']
+
+
+class ActionEnum(BaseModel):
+    __root__: Literal[
+        'add-spec',
+        'add-schema',
+        'add-snapshot',
+        'add-sort-order',
+        'add-view-version',
+        'assign-uuid',
+        'remove-partition-specs',
+        'remove-partition-statistics',
+        'remove-properties',
+        'remove-snapshot-ref',
+        'remove-snapshots',
+        'remove-statistics',
+        'set-current-schema',
+        'set-current-view-version',
+        'set-default-sort-order',
+        'set-default-spec',
+        'set-location',
+        'set-partition-statistics',
+        'set-properties',
+        'set-snapshot-ref',
+        'set-statistics',
+        'upgrade-format-version',
+    ]
+
+
 class BaseUpdate(BaseModel):
-    action: str
+    action: ActionEnum
 
 
 class AssignUUIDUpdate(BaseUpdate):
     """
     Assigning a UUID to a table/view should only be done when creating the table/view. It is not safe to re-assign the UUID if a table/view already has a UUID assigned
     """
 
-    action: Literal['assign-uuid']
+    action: ActionEnum
     uuid: str
 
 
 class UpgradeFormatVersionUpdate(BaseUpdate):
-    action: Literal['upgrade-format-version']
+    action: ActionEnum
```

Comment:
From a Pydantics' perspective this has a different semantic meaning. Before, we required that the `action` field for the `UpgradeFormatVersionUpdate` needs to be `upgrade-format-version`, now it can be any value from the enum. But, this can also be a limitation of the Pydantic open-api generator. Let me take a deeper look.

Reply:
For the Jersey one, it is the other way around, it will generate an `ActionEnum` per `Update`:

```java
public enum ActionEnum {
  ADD_SPEC(String.valueOf("add-spec"));

  private String value;

  ActionEnum(String value) {
    this.value = value;
  }

  @JsonValue
  public String getValue() {
    return value;
  }

  @Override
  public String toString() {
    return String.valueOf(value);
  }

  @JsonCreator
  public static ActionEnum fromValue(String value) {
    for (ActionEnum b : ActionEnum.values()) {
      if (b.value.equals(value)) {
        return b;
      }
    }
    throw new IllegalArgumentException("Unexpected value '" + value + "'");
  }
}
```

Reply:
@VladimirYushkevich @ajantha-bhat 

I've been playing around with alternative solutions. In Open-API there are many ways to get to the same result. What about https://github.com/apache/iceberg/pull/11955? The issue is related to the Enum, and with Open-API 3.1 the `const` is added looks more appropriate here. LMKWYT

Reply:
@Fokko if #11955 fixes java compilation issues I'm fine with that. I can close this PR but need to verify my generated java classes.